### PR TITLE
NVDAObjects.IAccessible: output alerts in braille

### LINKED ISSUE #14562: Report status in Braille when downloading files in Chrome
### Is your feature request related to a problem? Please describe.

As reported by Sergio via email, the status of files is reported in speech, but not in Braille when download files in Chrome.  Having the ability to have this information reported in Braille would be useful for those who rely on Braille either exclusively, or more than speech.

### Describe the solution you'd like

Report that the download is starting and when it is complete, and possibly status in between as needed, as is done in speech.

### Describe alternatives you've considered

I'm not sure if this was deliberately omitted to keep the Braille display uncluttered with constant messages, but if that is the case, perhaps some way of choosing which messages are reported?

### Additional context
I tested using:
NVDA 2022.4.0.27401 
Windows 11 (64-bit) Version: 22H2, Build: 22621.1105 
Chrome Version: 108.0.5359.126 

I noted that Microsoft Edge, Version: 109.0.1518.55 approaches it slightly differently, displaying a message when a file is downloaded and this is reported.

### PR DESCRIPTION
**Pull request by josephsl**
### Link to issue number:
Fixes #14562 

### Summary of the issue:
Alert event text is not shown on braille displays when the event occurs such as when downloading files via Google Chrome.

### Description of user facing changes
NVDA will present alert messages in braille.

### Description of development approach
Add braille output (braille.handler.message) to NVDAObjects.IAccessible.IAccessible.event_alert, borrowing heavily from notification behavior.

### Testing strategy:
Manual testing:

1. Visit websites such as community add-ons website (addons.nvda-project.org) where files can be downloaded.
2. Download a file while using a braille display.

Before the PR: speech output only when alert event is fired.
After the PR: speech and braille outpu ocurs.

### Known issues with pull request:
While Chrom's file download alert provides a good test case for testing braille output for alert event,s it might be possible that apps and web elements may not fire alert events, or if they do, presented in a way that NVDA doesn't know about (alert events are handled if the control role is alert).

### Change log entries:
Bug fixes:

In web browsers such as Chrome and Firefox, alerts such as file downloads via Chrome are shown in braille in addition to being spoken. (#14562)

### Code Review Checklist:

- [x] Pull Request description:
  - description is up to date
  - change log entries
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] API is compatible with existing add-ons.
- [x] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] Security precautions taken.

### CODE REVIEW

**Comment by josephsl** [comment]
Hi,

I think this might be a good opportunity to introduce a generic alert event to base NVDA object or the notification behavior (perhaps in 2023.x or 2024.1) unless alert event is specific to IAccessible and/or UIA. Among other things, the default implementation would allow vision enhancement providers to offer a way to look at where the alert is coming from.

Thanks.

File: source/NVDAObjects/IAccessible/__init__.py [hunk 1]
```diff
@@ -1702,9 +1702,12 @@ def event_alert(self):
 		if self in api.getFocusAncestors():
 			return
 		speech.speakObject(self, reason=controlTypes.OutputReason.FOCUS, priority=speech.Spri.NOW)
+		# Ideally, we wouldn't use getPropertiesBraille directly.
```
Comment (seanbudd) [inline source/NVDAObjects/IAccessible/__init__.py:1705]: what would be better? why not do that? it would be good to include this in the comment.

**Comment by seanbudd** [comment]
>I think this might be a good opportunity to introduce a generic alert event to base NVDA object or the notification behavior (perhaps in 2023.x or 2024.1) unless alert event is specific to IAccessible and/or UIA. Among other things, the default implementation would allow vision enhancement providers to offer a way to look at where the alert is coming from.

Would you be interested in doing this in this PR?

File: source/NVDAObjects/IAccessible/__init__.py (hunk 1 above)
Comment (seanbudd) [reply source/NVDAObjects/IAccessible/__init__.py:1705]: it appears that using this directly is common practice, I think this comment can be droppped

**Comment by josephsl** [comment]
Hi, not in this PR as this addresses a specific alert case (UIA is handled through system alert event); I think I’m seeing a mentoring opportunity for someone. Thanks for the suggestion (will think about it in northern summer once spring classes are done).

**Review by seanbudd — APPROVED** [review]

**Pull request merged**